uwblueprint/supportive-housing
# Tags Default Sort By Last Modified

## Notion task link


[Tags Default Sort By Last Modified](https://www.notion.so/uwblueprintexecs/Tags-Default-Sort-By-Last-Modified-1903efa76eea48219081c49d4d15b446)



## Implementation description
* Added a column 'last_modified' to the tags table to automatically record timestamps for create/update operations
* Modified GET route to return tags in order from most to least recently modified



## Steps to test
1. Upgrade database to latest migration: bash ./scripts/flask-db-upgrade.sh
2. In order to test these routes with Postman, first comment out @require_authorization_by_role({"Admin"}) on lines 10, 24, 47, and 71 of backend/app/rest/tags_routes.py.
3. Create a few tags by POSTing to /tags/ (e.g. {"name": “Cool Tag”, "status": "Active”}). 
4. Try GETting from /tags/. The tags you added should be returned in reverse chronological order.
5. Update a tag by PUTting to /tags/[id] (e.g. {"name": “Edited Tag”, "status": "Active”}). 
6. Try GETting from /tags/. The tag you just updated should be the first one returned (and the rest of the tags should also be returned in reverse chronological order).
7. Create & update a few more tags, and check that GETting them still results in the expected order.



## What should reviewers focus on?
* 


## Checklist
